scripts/common.py

```diff
@@ -784,3 +783,27 @@ def swap_env(key, value):
             del os.environ[key]
         else:
             os.environ[key] = old_value
+
+
+def write_stdout_safe(string):
+    """Tries to write the input string to stdout in a non-blocking way.
+
+    https://stackoverflow.com/a/44961052/4859885
+
+    Args:
+        string: str|bytes. The string to write to stdout.
+    """
+    string_bytes = string.encode('utf-8') if isinstance(string, str) else string
+
+    written = 0
+    while written < len(string_bytes):
+        try:
+            written += os.write(sys.stdout.fileno(), string_bytes[written:])
+        except io.UnsupportedOperation:
```

Comment:
What's going on in this case? Maybe add a comment explaining.

Reply:
`os.write` might not be supported in some cases. Added a comment.